### ISSUE TITLE
Fido.search errors very badly when no a.Time is specified

### ISSUE THREAD

**Issue by Cadair** [sunpy/sunpy#2992]
I am 90% sure this error is triggered because no `a.Time` is specified. If that is indeed the case the error that is raised is completely unhelpful.

```python
In [23]: Fido.search(a.Instrument("soon"))
```
```                                                                                                                                                                                           
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-23-e4b5b64d93c7> in <module>
----> 1 Fido.search(a.Instrument("soon"))

~/Git/sunpy/sunpy/net/fido_factory.py in search(self, *query)
    295         """  # noqa
    296         query = attr.and_(*query)
--> 297         return UnifiedResponse(query_walker.create(query, self))
    298 
    299     def fetch(self, *query_results, path=None, max_conn=5, progress=True,

~/Git/sunpy/sunpy/net/attr.py in create(self, *args, **kwargs)
    409 
    410     def create(self, *args, **kwargs):
--> 411         return self.createmm(self, *args, **kwargs)
    412 
    413     def apply(self, *args, **kwargs):

~/Git/sunpy/sunpy/util/multimethod.py in __call__(self, *args, **kwargs)
    119                 self.cache[types] = fun
    120                 return fun(*args, **kwargs)
--> 121         raise TypeError('{0!r}'.format(types))
    122 
    123     # XXX: Other Python implementations.

TypeError: (<class 'sunpy.net.vso.attrs.Instrument'>,)
```

**Comment by darthgera123** [comment]
i would like to work on this issue
could you please show how to recreate it

**Comment by nabobalis** [comment]
The code (minus the imports needed is at the top).

**Comment by Raahul-Singh** [comment]
I get a different error : 
```
TypeError: <Instrument('soon')> or any of its parents have not been registered with the AttrWalker
```
Could you tell me what the expected solution would be?
A better error message provided it indeed is caused when no `a.Time` is specified?

**Comment by Cadair** [comment]
I am actually fixing (or rather completely circumventing) this in #3770 so it's probably not worth spending time on.

**Comment by Raahul-Singh** [comment]
Cool.
I'll fish for work somewhere else. :grin:

**Comment by dstansby** [comment]
Is this fixed? I'm getting
```python
      5 res = Fido.search(a.jsoc.Series('hmi.synoptic_mr_polfil_720s'),
      6                   a.jsoc.PrimeKey('CAR_ROT', '2215'),
----> 7                   a.jsoc.Notify('sunpy@sunpy.org'))  

~/github/sunpy/sunpy/net/fido_factory.py in search(self, *query)
    318         """  # noqa
    319         query = attr.and_(*query)
--> 320         return UnifiedResponse(query_walker.create(query, self))
    321 
    322     def fetch(self, *query_results, path=None, max_conn=5, progress=True,

~/github/sunpy/sunpy/net/attr.py in create(self, *args, **kwargs)
    457         Call the create function(s) matching the arguments to this method.
    458         """
--> 459         return self.createmm(self, *args, **kwargs)
    460 
    461     def apply(self, *args, **kwargs):

~/github/sunpy/sunpy/util/functools.py in wrapper(*args, **kwargs)
     17 
     18     def wrapper(*args, **kwargs):
---> 19         return dispatcher.dispatch(args[1].__class__)(*args, **kwargs)
     20 
     21     wrapper.dispatch = dispatcher.dispatch

~/github/sunpy/sunpy/net/fido_factory.py in _create_and(walker, query, factory)
    249         for at in query.attrs:
    250             error += str(at) + ', '
--> 251         raise ValueError(error)
    252 
    253     # Return the response and the client

ValueError: The following part of the query did not have a time specified:
<Series('hmi.synoptic_mr_polfil_720s')>, <PrimeKey('CAR_ROT','2215')>, <Notify('sunpy@sunpy.org')>, 
```
which seems like quite a nice error message.

**Comment by Cadair** [comment]
Let's just called it closed by #3770 lol
